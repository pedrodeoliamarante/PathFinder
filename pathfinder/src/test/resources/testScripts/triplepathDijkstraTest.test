# Find a path with Dijkstra's algorithm when
# there are three different paths with different costs

CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D
AddNode simpleGraph E

AddEdge simpleGraph A B 10.0
AddEdge simpleGraph A C 1.0
AddEdge simpleGraph A D 3.0
AddEdge simpleGraph B E 4.0
AddEdge simpleGraph C E 5.0
AddEdge simpleGraph D E 6.0

FindPath simpleGraph A B

FindPath simpleGraph A C

FindPath simpleGraph A D

FindPath simpleGraph A E

FindPath simpleGraph B E

FindPath simpleGraph C E

FindPath simpleGraph D E
