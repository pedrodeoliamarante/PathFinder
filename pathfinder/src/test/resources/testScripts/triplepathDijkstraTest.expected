# Find a path with Dijkstra's algorithm when
# there are three different paths with different costs

created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph
added node E to simpleGraph

added edge 10.000 from A to B in simpleGraph
added edge 1.000 from A to C in simpleGraph
added edge 3.000 from A to D in simpleGraph
added edge 4.000 from B to E in simpleGraph
added edge 5.000 from C to E in simpleGraph
added edge 6.000 from D to E in simpleGraph

path from A to B:
A to B with weight 10.000
total cost: 10.000

path from A to C:
A to C with weight 1.000
total cost: 1.000

path from A to D:
A to D with weight 3.000
total cost: 3.000

path from A to E:
A to C with weight 1.000
C to E with weight 5.000
total cost: 6.000

path from B to E:
B to E with weight 4.000
total cost: 4.000

path from C to E:
C to E with weight 5.000
total cost: 5.000

path from D to E:
D to E with weight 6.000
total cost: 6.000